src/main/java/com/tarento/upsmf/examsAndAdmissions/service/ExamCycleService.java

```diff
@@ -444,5 +445,9 @@ public ResponseEntity<ResponseDto> processBulkUpload(MultipartFile file, String
             return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(response);
         }
     }
-
+    public List<ExamCycle> getExamCyclesByExamCycleAndCourse(Long instituteId) {
+        List<Course> courses = courseRepository.findAllByInstituteId(instituteId);
+        LocalDate currentDate = LocalDate.now();
+        return repository.findByCourseInAndStartDateAfter(courses, currentDate);
```

Comment:
the method should be findByCourseInAndEndDateAfter as we are interested in the exmcycles for which end data is today or after today